### PR TITLE
Add docker image and k8s configurations for testing on Travis

### PR DESCRIPTION
**Pull request by terrytangyuan**
Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

**What this PR does / why we need it**:

Part of #17. This PR adds necessary Dockerfile, k8s configurations. Note that the result of ` kubectl apply -f deploy/ftlib-test.yaml` won't be verified here yet. We'll add follow up PR to check the deployment status.

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: deploy/ftlib-test.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      containers:
+        - image: ftlib:deploy
+          imagePullPolicy: Never
+          name: ftlib-test-c
+          ports:
+          - containerPort: 3000
+            name: http
+          - containerPort: 22
+            name: ssh
+          command: ["python3", "/ftlib/test/generic_test.py"]
```
Comment (zw0610) [inline deploy/ftlib-test.yaml]: Will this be sufficient for the deploy test? But honestly, I still cannot figure out a way to test, specially given we are building something 'fault-tolerant'.

File: deploy/ftlib-test.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          resources:
```
Comment (zw0610) [inline deploy/ftlib-test.yaml]: The `dummy_NCCL` needs GPUs for communication, which seems not budget-friendly.

File: deploy/ftlib-test.yaml (hunk 1 above)
Comment (terrytangyuan) [reply deploy/ftlib-test.yaml]: Good call. I updated this to pytorch test. We haven't added the validation logic around this yet. Testing the fault-tolerant part can be a bit tricky that requires some thoughts.

File: deploy/ftlib-test.yaml (hunk 2 above)
Comment (terrytangyuan) [reply deploy/ftlib-test.yaml]: Travis doesn't have GPUs for testing. I updated this command to run pytorch test instead.

**Pull request merged**